functions/src/glean.ts

```diff
@@ -0,0 +1,228 @@
+import Glean from "@mozilla/glean/node";
+import PingEncryptionPlugin from "@mozilla/glean/plugins/encryption";
+import axios from "axios";
+import { Mutex, Semaphore, withTimeout } from "async-mutex";
+
+import * as rallyMetrics from "./generated/rally.js";
+import * as userMetrics from "./generated/user.js";
+import * as enrollmentMetrics from "./generated/enrollment.js";
+import * as unenrollmentMetrics from "./generated/unenrollment.js";
+import * as rallyPings from "./generated/pings.js";
+
+import {
+  Uploader,
+  UploadResult,
+  UploadResultStatus,
+} from "@mozilla/glean/uploader";
+
+const GLEAN_DEBUG_VIEW_TAG = "MozillaRally";
+const GLEAN_RALLY_APP_ID = process.env.NODE_ENV !== "test" ? "rally-core" : "test-app-id";
```

Comment:
@rhelmer this code to prevents Glean pings from getting sent to the "real" rally-core app during `npm run test`, but I'm wondering if we should do something about the `npm run dev` case, i.e. when we're all doing general local testing? Is it ok for those pings to get sent to the main environment, or should we disable/redirect them? I'm not clear on how that all currently works in the Core Addon...

Reply:
The core add-on just doesn't initialize Glean at all in dev mode:
https://github.com/mozilla-rally/rally-core-addon/blob/30ccc1ea49046fd188c06a8e884d846de0a0037c/core-addon/DataCollection.js#L39

Reply:
Decided to disable Glean by default whenever the Firebase Emulator is running, with an option to explicitly enable it via an `ENABLE_GLEAN` environment variable. When the variable is set the pings are logged and sent to a test app id. When running in the "real" non-emulator environment, pings aren't logged and are sent to the "rally-core" app id. Added some info to the README. Lmk your thoughts.